fix: Removes update logic for the global_cluster_config resource. (#2282)
mongodb/terraform-provider-mongodbatlas: website/docs/r/global_cluster_config.html.markdown

-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.

> ~> **IMPORTANT:** A Global Cluster Configuration, once created, can only be deleted and never re-created again with the same data if not without the help of the Atlas UI. This is because the configuration and its related collection with shard key and indexes are managed separately and they would end up in an inconsistent state. [Read more about Global Cluster Configuration](https://www.mongodb.com/docs/atlas/global-clusters/)

Reviewer: i guess we can't prevent the resource for being created again once deleted?  it'd be nice if we could detect that conditions and block the create (the second time)

Author: yes, but I can't think of a way on our side. It should be something the Atlas itself need to implement a block at API level.

> ~> **IMPORTANT:** A Global Cluster Configuration, once created, can only be deleted and never re-created again with the same data if not without the help of the Atlas UI. This is because the configuration and its related collection with shard key and indexes are managed separately and they would end up in an inconsistent state. [Read more about Global Cluster Configuration](https://www.mongodb.com/docs/atlas/global-clusters/)

Reviewer 2: Should we include about this in the release guide too? 
Not sure if this should be considered a breaking change, I understand we are removing an incorrect behavior (just thinking out loud)

Author: It should not be considered a breaking change because it's a bug. I wasn't planning to put it in the release guide, but let me know otherwise

> ~> **IMPORTANT:** A Global Cluster Configuration, once created, can only be deleted and never re-created again with the same data if not without the help of the Atlas UI. This is because the configuration and its related collection with shard key and indexes are managed separately and they would end up in an inconsistent state. [Read more about Global Cluster Configuration](https://www.mongodb.com/docs/atlas/global-clusters/)

Reviewer 3: Suggestion: the wording is confusing here, suggest we clarify it:
```suggestion
~> **IMPORTANT:** A Global Cluster Configuration, once created, can only be deleted. You can recreate the Global Cluster with the same data only in the Atlas UI. This is because the configuration and its related collection with shard key and indexes are managed separately and they would end up in an inconsistent state. [Read more about Global Cluster Configuration](https://www.mongodb.com/docs/atlas/global-clusters/)
```
